### ISSUE TITLE
babel 7

### ISSUE THREAD

**Issue by chrusart** [ttag-org/ttag-cli#43]
@AlexMost can you predict when it's gonna be live?
I just finished as well babel 7 switch and was going to make a PR, but you're faster :)
Anyway, I'm not babel expert but after upgrading to babel 7 actually I had to add more plugins to extract translations.
I'm a little bit confused as I could extract translations before using ttag-cli, even if I am using decorators, class properties etc. in code.
According to https://github.com/ttag-org/ttag-cli/issues/38 there will be an option to add extra plugins or I missunderstood the idea?

**Comment by chrusart** [comment]
My ```src/lib/extract.ts```
```
    const babelOptions = {
        presets: [
          babelPresetReact,
          babelPresetFlow,
        ],
        plugins: [
          babelPluginRestSpread,
          [babelPluginDecorators, { "legacy": true }],
          babelPluginClassProperties,
          [babelPluginTtag, c3pOptions],
        ]
    };
```

**Comment by chrusart** [comment]
When you release I will test right away if it's working for my react-native project.

**Comment by AlexMost** [comment]
I have just published pre minor version `1.3.0-0` with babel 7. Not sure about how it will work with the decorators. But in your case, I think, we definitely need to implement #38 to make that work. Going to do that ASAP.

**Comment by chrusart** [comment]
Yeah, just noticed, my bad, didn't look in npm repo, just in github releases. Will test right now.

**Comment by chrusart** [comment]
@AlexMost is ```dist``` built with babel 7?
Clean yarn install and I get babel versions mismatch:
```
⠋ [ttag] extracting translations to translations.pot ...{ Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
```

but if I build node_modules/ttag-cli then it works (I still have errors about classProperties etc. but it's another thing we talked just few minutes ago)

**Comment by chrusart** [comment]
Ok, for me to work, this is what i need to add (at least to extract):
```package.json```
```
    "@babel/preset-flow": "^7.0.0",
    "@babel/plugin-proposal-class-properties": "^7.0.0",
    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
    "@babel/plugin-proposal-decorators": "^7.0.0",
```
```declarations.ts```
```
declare module "@babel/preset-flow";
declare module '@babel/plugin-proposal-class-properties';
declare module '@babel/plugin-proposal-object-rest-spread';
declare module '@babel/plugin-proposal-decorators';
```
```src/lib/extract.ts```
```
import * as babelPresetFlow from "@babel/preset-flow";
import babelPluginClassProperties from "@babel/plugin-proposal-class-properties";
import babelPluginRestSpread from "@babel/plugin-proposal-object-rest-spread";
import babelPluginDecorators from "@babel/plugin-proposal-decorators";
...
    const babelOptions = {
        presets: [babelPresetReact, babelPresetFlow],
        plugins: [
            babelPluginRestSpread,
            [babelPluginDecorators, { legacy: true }],
            babelPluginClassProperties,
            [babelPluginTtag, c3pOptions]
        ]
    };
```

But weird thing is that I can add this stuff to node_modules/ttag-cli in my project, build it, and it works.
But if I add it to cloned git repo I got bunch of other errors (which I resolved locally when upgrading to babel 7)

**Comment by AlexMost** [comment]
please, try `1.3.0-1` version, It should work if you will configure all your plugins inside your projects `.babelrc` config.

**Comment by chrusart** [comment]
@AlexMost  Version mismatch is resolved, thanks!. Is it gonna harm if we add my PR with some other plugins?

**Comment by AlexMost** [comment]
@chrusart my opinion is that we should not hardcode plugins inside the cli tool and provide the ability for developers to configure their build process. I am going to remove `@babel/preset-react` also.

**Comment by AlexMost** [comment]
I have set up an example repository with `.babelrc` that adds decorators support - https://github.com/ttag-org/ttag-cli-babel-7-demo
I have checked those examples in a couple of environments and seems like they are working, can you check them?

**Comment by chrusart** [comment]
Sure @AlexMost, I totally agree, I'm not experienced in babel at all, but it was clear for me that it's silly too that cli must have all plugins that maybe ever gonna be needed. I will close PR, just made it to see clearly what works for me and maybe give some clue.
